tests/UnifiedSpecTests/UnifiedSpecTest.php

```diff
@@ -141,6 +141,19 @@ private function doTestCase(stdClass $test, string $schemaVersion, array $runOnR
         }
     }
 
+    /**
+     * @dataProvider provideCrudTests
+     */
+    public function testCrud(...$args)
```

Comment:
Having separate test methods for individual specs allows for easier filtering of what tests to execute (without having to mess with `--filter` and dataset names.

Reply:
Is this change related to anything else in the PR? I don't see any tests added to the `crud/` directory.

Given that UnifiedSpecTest is primarily testing its own tests (e.g. passing/failing), I wonder if other specs using the runner should just have their own test classes.

This would probably be best deferred until after I merge #806, as that PR extracts the runner itself from the test class. If this isn't urgent, I'd suggest pulling this out to avoid merge conflicts and just proposing this in a PHPLIB ticket dependent on PHPLIB-613.

Reply:
Given that the tests for each spec would only contain a single `test<Spec>(...$args)` method, I'm not sure what there is to be gained by having each in a separate class. If we want to separate them somewhat, I'd suggest the following:
* Refactor `UnifiedSpecTest` to an abstract `UnifiedSpecTestCase` class
* Add a `UnifiedRunnerTest` class that only tests the runner itself (e.g. passing/failing)
* From there, add either a common `SpecTest` class or a single `<Spec>Test` class for each spec that has tests in the unified format.

In the meantime, I can extract this change to a separate file until we're done with #806, especially considering that the PR for PHPLIB-622 will add versioned API tests where we'll see the problem. That should avoid merge conflicts in the refactoring PR.

Reply:
To follow up on the tests themselves, there's a single new test file added to `tests/UnifiedSpecTests/crud/`, which is what is tested here. The other tests have not been ported to the unified format yet.

Thinking of it, I remember discussing where these tests would live, but don't remember coming to a conclusion. I've put them in the `UnifiedSpecTests` folder for the time being, but we may want to not use that directory as a catch-all for unified spec tests and rather use different directories for it.

Reply:
Oh, I must have missed `tests/UnifiedSpecTests/crud/estimatedDocumentCount.json` in the previous review. I see it now.